### ISSUE TITLE
Unary gRPC drops trace-context propagation when client tracing is disabled

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#6708]
Unconfirmed.  This is the result of a search with Claude, may be a false positive.

In `src/gax-internal/src/grpc.rs`, unary gRPC requests inject distributed trace context into outbound metadata headers:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/gax-internal/src/grpc.rs#L496-L506

The injected span is created by `grpc_helpers::unary_make_request_span`. When `self.tracing_attributes` is `None` (client telemetry tracing disabled), `unary_make_request_span` returns `tracing::Span::none()`:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/gax-internal/src/grpc/grpc_helpers.rs#L142-L144

Injecting from `Span::none()` sends no traceparent or baggage headers. In contrast, all other transport paths (HTTP at `http.rs:326-331`, `501-506`, and gRPC streaming at `grpc.rs:212`, `367`) inject from `&tracing::Span::current()` to propagate active trace context regardless of whether client tracing is enabled.

**Comment by SchahinRohani** [comment]
I added a local unary gRPC integration test based on `propagate_trace_context` in `src/gax-internal/tests/grpc_observability.rs` (not yet submitted as a PR).

With `config.tracing = false`, I executed the request under an active `tracing` parent span. The test checks that the parent span context is valid and that the `traceparent` received by the echo server contains the parent's trace ID.

It passes with GOOGLE_CLOUD_RUST_LOGGING=false:

```sh
GOOGLE_CLOUD_RUST_LOGGING=false cargo test -p google-cloud-gax-internal \
  --test grpc_observability \
  propagate_trace_context_without_client_tracing -- --nocapture
```

Based on `tracing_enabled()` and `make_inner()`, this should leave `tracing_attributes` as `None`, causing `unary_make_request_span()` to return `Span::none()`. Nevertheless, the parent trace ID reaches the server in this setup. I haven't determined which context path accounts for the successful propagation.

Is there a different setup in which you expect this to fail? If useful, I can submit the test as a regression test.
